### PR TITLE
[ROCm] clang support

### PR DESCRIPTION
**Pull request by Ruturaj4**
Brings up clang support for ROCm XLA. This is required to implement clang support for JAX+XLA for ROCm.

@draganmladjenovic @i-chaochen

### CODE REVIEW

**Comment by i-chaochen** [comment]
Hi @xla-rotation could you have a look, please! thanks!

File: build_tools/configure/configure.py
Comment (ddunl) [inline build_tools/configure/configure.py:375]: I think this line needs deleting, other than that LGTM

**Comment by Ruturaj4** [comment]
@ddunl lets hold off upstreaming this, we need some testing to do. I will ping you once it is done.

**Comment by Ruturaj4** [comment]
@ddunl can you review/ merge now please. I have very few new changes particularly in `BUILD.rocm.tpl` and `crosstool_wrapper_driver_rocm.tpl`. Thanks!

**Review by ddunl — APPROVED** [review]

**Comment by ddunl** [comment]
Made a couple extra changes to fix configure_test on ARM, should merge later today or Monday. Sorry for the delay!

**Pull request closed without merging**
